contracts/contracts/strategies/amo/BalancerEthAMOStrategy.sol

```diff
@@ -0,0 +1,135 @@
+import { IERC4626 } from "../../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+contract BalancerEthAMOStrategy is BaseBalancerAMOStrategy {
+    using StableMath for uint256;
+
+    constructor(
+        BaseStrategyConfig memory _baseConfig,
+        AMOConfig memory _amoConfig,
+        BalancerConfig memory _balancerConfig
+    ) BaseBalancerAMOStrategy(_baseConfig, _amoConfig, _balancerConfig) {}
+
+    /***************************************
+        Vault to Pool Asset Conversions
+    ****************************************/
+
+    /// @dev WETH is the Vault asset and the Balancer pool asset so
+    /// nothing to except return the vault asset amount
+    function _toPoolAsset(address, uint256 assets)
+        internal
+        pure
+        override
+        returns (uint256 poolAssets)
+    {
+        poolAssets = assets;
+    }
+
+    function _calcPoolAsset(address, uint256 vaultAssetAmount)
+        internal
+        pure
+        override
+        returns (uint256 poolAssetAmount)
+    {
+        poolAssetAmount = vaultAssetAmount;
+    }
+
+    /// @dev WETH is the Vault asset and the pool asset so return the WETH amount
+    /// @param poolAssetAmount Amount of WETH to convert to OETH
+    /// @param oTokenAmount Amount of OETH converted from WETH
+    function _toOTokens(uint256 poolAssetAmount)
+        internal
+        pure
+        override
+        returns (uint256 oTokenAmount)
+    {
+        oTokenAmount = poolAssetAmount;
+    }
+
+    /***************************************
+            Curve Pool Withdrawals
+    ****************************************/
+
+    /// @dev transfers the specified WETH amount to the recipient
+    function _withdrawAsset(
+        address,
+        uint256 vaultAssetAmount,
+        address _recipient
+    ) internal override {
+        // Transfer the WETH to the Vault
+        require(
+            asset.transfer(_recipient, vaultAssetAmount),
+            "WETH transfer failed"
+        );
+
+        emit Withdrawal(address(asset), address(lpToken), vaultAssetAmount);
+    }
+
+    /// @dev transfers the EETH balance of this strategy contract to the recipient
+    function _withdrawAllAsset(address _recipient) internal override {
+        uint256 vaultAssets = asset.balanceOf(address(this));
+
+        _withdrawAsset(address(asset), vaultAssets, _recipient);
+    }
+
+    /***************************************
+                Asset Balance
+    ****************************************/
+
+    /**
+     * @notice Get strategy's share of an assets in the Balancer pool.
+     * This is not denominated in OUSD/ETH value of the assets in the Balancer pool.
+     * @param _asset  Address of the Vault asset. eg WETH
+     * @return balance  the amount of vault assets
+     *
+     * IMPORTANT if this function is overridden it needs to have a whenNotInBalancerVaultContext
```

Comment:
🔴 We are missing the whenNotInBalancerVaultContext modifier on this function?

Personally, I'd rather it be inline, than a modifier?

Reply:
I will add that in. Sure making it inline, for better readability